aliyun/src/main/java/org/apache/iceberg/aliyun/oss/BaseOSSFile.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aliyun.oss;
+
+import com.aliyun.oss.OSS;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+
+abstract class BaseOSSFile {
+  private final OSS client;
+  private final OSSURI uri;
+
+  BaseOSSFile(OSS client, OSSURI uri) {
+    this.client = client;
+    this.uri = uri;
+  }
+
+  public String location() {
+    return uri.location();
+  }
+
+  public OSS client() {
+    return client;
+  }
+
+  public OSSURI uri() {
+    return uri;
+  }
+
+  public boolean exists() {
+    return client.doesObjectExist(uri.bucket(), uri.key());
```

Comment:
Is there a way to combine this with `getLength`? I think in other places we fetch a "stat" object so that we don't need to do separate requests.